diff -u -r result/src/expr/expr_template.h git-export/src/expr/expr_template.h
--- result/src/expr/expr_template.h	1969-12-31 19:00:01.000000000 -0500
+++ git-export/src/expr/expr_template.h	2014-11-01 09:15:48.000000000 -0400
@@ -649,7 +649,7 @@
     long& l = out.iword(s_iosIndex);
     if(l == 0) {
       // set the default print depth on this ostream
-      if(&Options::current() != NULL) {
+      if(Options::hasCurrent()) {
         l = options::defaultExprDepth();
       }
       if(l == 0) {
@@ -801,7 +801,7 @@
     if(l == 0) {
       // set the default dag setting on this ostream
       // (offset by one to detect whether default has been set yet)
-      if(&Options::current() != NULL) {
+      if(Options::hasCurrent()) {
         l = options::defaultDagThresh() + 1;
       }
       if(l == 0) {
@@ -884,7 +884,7 @@
     if(l == 0) {
       // set the default language on this ostream
       // (offset by one to detect whether default has been set yet)
-      if(&Options::current() != NULL) {
+      if(Options::hasCurrent()) {
         l = options::outputLanguage() + 1;
       }
       if(l <= 0 || l > language::output::LANG_MAX) {
diff -u -r result/src/options/options.h git-export/src/options/options.h
--- result/src/options/options.h	1969-12-31 19:00:01.000000000 -0500
+++ git-export/src/options/options.h	2014-11-01 09:14:29.000000000 -0400
@@ -65,6 +65,10 @@
     return *s_current;
   }
 
+  static inline bool hasCurrent() {
+    return s_current == NULL;
+  }
+
   Options();
   Options(const Options& options);
   ~Options();
diff -u -r result/src/printer/printer.h git-export/src/printer/printer.h
--- result/src/printer/printer.h	1969-12-31 19:00:01.000000000 -0500
+++ git-export/src/printer/printer.h	2014-11-01 09:16:16.000000000 -0400
@@ -61,7 +61,7 @@
       //
       // Options can be null in certain circumstances (e.g., when printing
       // the singleton "null" expr.  So we guard against segfault
-      if(&Options::current() != NULL) {
+      if(Options::hasCurrent()) {
         if(options::outputLanguage.wasSetByUser()) {
           lang = options::outputLanguage();
         }
